androidClient/app/src/main/java/com/csatimes/dojma/adapters/ContributorsAdapter.java

```diff
@@ -13,26 +13,89 @@
  * Created by vikramaditya on 24/2/17.
  */
 
-public class ContributorsAdapter extends RecyclerView.Adapter<ContributorsViewHolder> {
+public class ContributorsAdapter extends RecyclerView.Adapter<ContributorsViewHolder> implements ValueEventListener {
 
-    private List<String> contributorsList;
+    private ArrayList<Contributor> contributors;
+    private Context context;
+    private Activity activity;
+    Exception e;
 
-    public ContributorsAdapter(List<String> contributorsList) {
-        this.contributorsList = contributorsList;
+    public ContributorsAdapter(Context context) {
+        this.context = context;
+        populate();
+        DatabaseReference devRef = FirebaseDatabase.getInstance().getReference()
+                .child(DHC.FIREBASE_DATABASE_REFERENCE_CONTRIBUTORS);
+        devRef.addValueEventListener(this);
+        devRef.keepSynced(true);
     }
 
     @Override
     public ContributorsViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
-        return new ContributorsViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_format_contributors, parent, false));
+        final LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+        final View contributor = inflater.inflate(R.layout.item_format_contributors, parent, false);
+        return new ContributorsViewHolder(contributor, context);
     }
 
     @Override
     public void onBindViewHolder(ContributorsViewHolder holder, int position) {
-        holder.nameTextView.setText(contributorsList.get(position));
+        holder.populate(contributors.get(position));
+
     }
 
     @Override
     public int getItemCount() {
-        return contributorsList.size();
+        return (contributors.size());
+    }
+
+    private void populate() {
+        contributors = new ArrayList<>();
+        contributors.add(new Contributor(
+                "Vikramaditya Kukreja",
```

Comment:
Please, no hard coding

Reply:
this is for the case when there is no internet connection. It is same as did in ard and as was said by tanmay

Reply:
we will enable firebase persistence, so offline will not be issue